app/utilities/supplementary_data_parser.py

```diff
@@ -27,54 +26,63 @@ class SupplementaryData(Schema, StripWhitespaceMixin):
     schema_version = VALIDATORS["string"](
         validate=validate.OneOf([SupplementaryDataSchemaVersion.V1.value])
     )
-    items = fields.Nested(ItemsData, required=False, unknown=EXCLUDE)
+    items = fields.Nested(ItemsData, required=False, unknown=INCLUDE)
 
     @validates_schema()
     def validate_unit_id(self, data, **kwargs):
         # pylint: disable=no-self-use, unused-argument
         if (
             data
             and data.get("identifier")
-            and data["identifier"] != self.context["ru_ref"]
+            and data["identifier"] != self.context["unit_id"]
         ):
             raise ValidationError(
                 "Supplementary data did not return the specified Unit ID"
             )
 
 
 class SupplementaryDataMetadataSchema(Schema, StripWhitespaceMixin):
-    dataset_id = VALIDATORS["string"](validate=validate.Length(min=1))
+    dataset_id = VALIDATORS["uuid"]()
     survey_id = VALIDATORS["string"](validate=validate.Length(min=1))
     data = fields.Nested(
         SupplementaryData,
         required=True,
-        unknown=EXCLUDE,
+        unknown=INCLUDE,
         validate=validate.Length(min=1),
     )
 
     @validates_schema()
-    def validate_dataset_id(self, data, **kwargs):
+    def validate_dataset_and_survey_id(self, data, **kwargs):
         # pylint: disable=no-self-use, unused-argument
-        if (
-            data
-            and data.get("dataset_id")
-            and data["dataset_id"] != self.context["dataset_id"]
-        ):
-            raise ValidationError(
-                "Supplementary data did not return the specified Dataset ID"
-            )
+        if data:
+            if (
+                data.get("dataset_id")
```

Comment:
I know it wasn't directly stated on the tech details, but do we need to return an error in the case that we have data with no dataset_id or survey_id? could just modify the .get if so

Reply:
Ah I see they actually can't be none, it looks like validation against length 1 and being a uuid has already happened so is this just for type hinting? would a type ignore be more suitable?

Reply:
Updated this now and removed the .get 👍